packages/usa-modal/src/index.js

```diff
@@ -278,51 +279,50 @@ const cleanUpModal = (baseComponent) => {
   modalWrapper.parentElement.removeChild(modalWrapper);
 };
 
-modal = {
-  init(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      const modalId = modalWindow.id;
-      setUpModal(modalWindow);
-
-      // this will query all openers and closers including the overlay
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => {
-          // Turn anchor links into buttons because of
-          // VoiceOver on Safari
-          if (item.nodeName === "A") {
-            item.setAttribute("role", "button");
-            item.addEventListener("click", (e) => e.preventDefault());
-          }
-
-          // Can uncomment when aria-haspopup="dialog" is supported
-          // https://a11ysupport.io/tech/aria/aria-haspopup_attribute
-          // Most screen readers support aria-haspopup, but might announce
-          // as opening a menu if "dialog" is not supported.
-          // item.setAttribute("aria-haspopup", "dialog");
-
-          item.addEventListener("click", toggleModal);
-        });
-    });
-  },
-  teardown(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      cleanUpModal(modalWindow);
-      const modalId = modalWindow.id;
-
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => item.removeEventListener("click", toggleModal));
-    });
-  },
-  focusTrap: null,
-  toggleModal,
-  on(root) {
-    this.init(root);
-  },
-  off(root) {
-    this.teardown(root);
-  },
-};
+modal = behavior(
+  {},
+  {
+    init(root) {
+      selectOrMatches(MODAL, root).forEach((modalWindow) => {
+        const modalId = modalWindow.id;
+        if (modalId) {
+          setUpModal(modalWindow);
+        }
+
+        // this will query all openers and closers including the overlay
+        document
+          .querySelectorAll(`[aria-controls="${modalId}"]`)
+          .forEach((item) => {
+            // Turn anchor links into buttons because of
+            // VoiceOver on Safari
```

Comment:
Is this statement correct? I thought this was for _all_ screen readers? You can also fit it all in one line.

Reply:
Resolved in [ee866c9](https://github.com/uswds/uswds/pull/5208/commits/ee866c9f19678cfb9b24c9161cd56f5b8f12285c)